mesh/meshdb.go

```diff
@@ -361,41 +366,39 @@ func (m *MeshDB) writeTransactions(blk *types.Block) ([]types.TransactionId, err
 	return txids, nil
 }
 
-func (m *MeshDB) getTransactions(transactions []types.TransactionId) ([]*types.SerializableTransaction, error) {
-	var ts []*types.SerializableTransaction
-	for _, id := range transactions {
-		tBytes, err := m.getTransactionBytes(id)
-
-		if err != nil {
-			m.Error("error retrieving transaction from database ", err)
-		}
-
-		t, err := types.BytesAsTransaction(tBytes)
+func (m *MeshDB) GetTransactions(transactions []types.TransactionId) (
+	ts map[types.TransactionId]*types.SerializableTransaction,
```

Comment:
why not return this as an array? the map values are copied to array anyway..

Reply:
i need to keep the order of the txs this make it easier

Reply:
I don't get it. if you want to keep the order then it's better that you'd use a slice from the beginning as a map has a random order.

Reply:
no, the slice dosen't contain all the ids, i use the map to merge the fetched transactions and the ones i already know.
i can do it from slices but than it would be much less efficient as i would need to look up each transaction in the slice